src/main/kotlin/com/sourcegraph/cody/agent/protocol/ProtocolTextDocument.kt

```diff
@@ -102,60 +104,63 @@ private constructor(
     fun fromEditorWithRangeSelection(editor: Editor, event: SelectionEvent): ProtocolTextDocument? {
       val file = FileDocumentManager.getInstance().getFile(editor.document) ?: return null
       val uri = uriFor(file)
-      val document = editor.document
-
-      val startOffset = event.newRange.startOffset
-      val startLine = document.getLineNumber(startOffset)
-      val lineStartOffset1 = document.getLineStartOffset(startLine)
-      val startCharacter = startOffset - lineStartOffset1
-
-      val endOffset = event.newRange.endOffset
-      val endLine = document.getLineNumber(endOffset)
-      val lineStartOffset2 =
-          if (startLine == endLine) {
-            lineStartOffset1
-          } else {
-            document.getLineStartOffset(endLine)
-          }
-      val endCharacter = endOffset - lineStartOffset2
-
-      val selection = Range(Position(startLine, startCharacter), Position(endLine, endCharacter))
+      val selection =
+          editor.document.codyRange(event.newRange.startOffset, event.newRange.endOffset)
       return ProtocolTextDocument(
           uri = uri,
           selection = selection,
           testing =
               getTestingParams(
                   uri = uri,
-                  content = document.text,
+                  content = editor.document.text,
                   selection = selection,
                   selectedText = editor.selectionModel.selectedText))
     }
 
+    private val isFullDocumentSyncEnabled =
+        System.getProperty("cody-agent.fullDocumentSyncEnabled") == "true"
+
     @JvmStatic
     @RequiresEdt
     fun fromEditorForDocumentEvent(editor: Editor, event: DocumentEvent): ProtocolTextDocument? {
-      val oldFragment = event.oldFragment.toString()
-      val file = FileDocumentManager.getInstance().getFile(editor.document) ?: return null
-      val startPosition = editor.offsetToLogicalPosition(event.offset).codyPosition()
```

Comment:
I think the root problem with `\t` was in `offsetToLogicalPosition`. I didn't confirm this but the "logical position" may have converted tabs to 2 character. There's nothing in the docstring indicating what the contract is.